### DISCUSSION TITLE
How to create reference for scoring

### DISCUSSION THREAD

**Discussion by johko** [explosion/spaCy#9152] [Help: Coding & Implementations] [answered]
Hi,
in v3 spacy dropped the GoldParser and 'replaced' it with Example. But I don't quite get how to create the reference that should be included in the Example object. The example in the doc shows:

```
from spacy.tokens import Doc
from spacy.training import Example

words = ["hello", "world", "!"]
spaces = [True, False, False]
predicted = Doc(nlp.vocab, words=words, spaces=spaces)
reference = parse_gold_doc(my_data)
example = Example(predicted, reference)
```

But I suppose this _parse_gold_doc_ is rather a dummy than a real function (at least I can't find it)

**Comment by polm** [comment, marked as answer]
The main thing about examples in v3 is that the `reference` doc is just what you'd like the output from your component to be. So if you're preparing NER training data, you just set the correct entity labels. For other components you set the correct labels for that component. There's an example of this in the [training data section](https://spacy.io/usage/training#training-data) of the docs.

When creating training data, you typically create a DocBin, which contains Docs and not Examples - when you give the DocBin to spaCy, it'll treat the contents as the `reference` for Examples and fill in the other doc itself.

Does that answer your question?

**Reply by johko** [reply]
I see, it wasn't clear to me that I can just use the DocBin, thank you

**Reply by polm** [reply]
No worries! We only just added this to the docs and are working on making it clearer. You might also find the [example projects](https://github.com/explosion/projects/tree/v3/tutorials) a useful reference - they have scripts that create `.spacy` files, some simple and some complex.
